haystack/modeling/utils.py

```diff
@@ -296,3 +301,153 @@ def all_gather_list(data, group=None, max_size=16384):
             'in your training script that can cause one worker to finish an epoch '
             'while other workers are still iterating over their portions of the data.'
         )
+
```

Comment:
In general I rearranged functions for the farm base components PR such that if they are only used inside a single script I moved them to the script.

Of course this now changes for some functions since they are used in the inferencer as well.
So can you please make sure the functions: 
calc_chunksize, log_ascii_workers and grouper are removed from data_silo.py and imported from this utils.py